src/database.c

```diff
@@ -419,25 +449,34 @@ _add_to_db(ProfMessage* message, char* type, const Jid* const from_jid, const Ji
 
         sqlite3_stmt* lmc_stmt = NULL;
 
-        if (SQLITE_OK == sqlite3_prepare_v2(g_chatlog_database, replace_check_query, -1, &lmc_stmt, NULL)) {
-            if (sqlite3_step(lmc_stmt) == SQLITE_ROW) {
-                const char* from_jid_orig = (const char*)sqlite3_column_text(lmc_stmt, 0);
+        if (SQLITE_OK != sqlite3_prepare_v2(g_chatlog_database, replace_check_query, -1, &lmc_stmt, NULL)) {
+            log_error("SQLite error in _add_to_db() on selecting original message: %s", sqlite3_errmsg(g_chatlog_database));
+            return;
+        }
+
+        if (sqlite3_step(lmc_stmt) == SQLITE_ROW) {
+            original_message_id = sqlite3_column_int(lmc_stmt, 0);
+            const char* from_jid_orig = (const char*)sqlite3_column_text(lmc_stmt, 1);
+
+            // Handle non-XEP-compliant replacement messages (edit->edit->original)
```

Comment:
Didn't check this part in detail. I remember @sjaeckel mentioned something about not fixing others bugs in the MUC.

Reply:
> Didn't check this part in detail. I remember @sjaeckel mentioned something about not fixing others bugs in the MUC.

It's also our bug as well. At the very least we should be compatible with older profanity versions.